types/chrome/index.d.ts

```diff
@@ -5815,46 +5815,81 @@ declare namespace chrome {
         /** An object encapsulating one visit to a URL. */
         export interface VisitItem {
             /** The transition type for this visit from its referrer. */
-            transition: string;
-            /** Optional. When this visit occurred, represented in milliseconds since the epoch. */
-            visitTime?: number | undefined;
+            transition: `${TransitionType}`;
+            /**
+             * True if the visit originated on this device. False if it was synced from a different device
+             * @since Chrome 115
+             */
+            isLocal: boolean;
+            /** When this visit occurred, represented in milliseconds since the epoch. */
+            visitTime?: number;
             /** The unique identifier for this visit. */
             visitId: string;
             /** The visit ID of the referrer. */
             referringVisitId: string;
-            /** The unique identifier for the item. */
+            /** The unique identifier for the corresponding {@link history.HistoryItem}. */
             id: string;
         }
 
         /** An object encapsulating one result of a history query. */
         export interface HistoryItem {
-            /** Optional. The number of times the user has navigated to this page by typing in the address. */
-            typedCount?: number | undefined;
-            /** Optional. The title of the page when it was last loaded. */
-            title?: string | undefined;
-            /** Optional. The URL navigated to by a user. */
-            url?: string | undefined;
-            /** Optional. When this page was last loaded, represented in milliseconds since the epoch. */
-            lastVisitTime?: number | undefined;
-            /** Optional. The number of times the user has navigated to this page. */
-            visitCount?: number | undefined;
+            /** The number of times the user has navigated to this page by typing in the address. */
+            typedCount?: number;
+            /** The title of the page when it was last loaded. */
+            title?: string;
+            /** The URL navigated to by a user. */
+            url?: string;
+            /** When this page was last loaded, represented in milliseconds since the epoch. */
+            lastVisitTime?: number;
+            /** The number of times the user has navigated to this page. */
+            visitCount?: number;
             /** The unique identifier for the item. */
             id: string;
         }
 
+        /**
+         * The transition type for this visit from its referrer.
+         * @since Chrome 44
+         */
+        export enum TransitionType {
```

Comment:
It's better to define this as an object literal type; TypeScript enums allow reverse-mapping, meaning they are defined at runtime like (simplified for readability):

```ts
var TransitionType = {
  LINK: "link",
  TYPED: "typed",
  // ...
  // Reverse mappings:
  link: "LINK",
  typed: "TYPED",
  // ...
}
```

but the Chrome API doesn’t have those reverse mappings:

<img width="349" height="239" alt="Image" src="https://github.com/user-attachments/assets/a3885e4f-2fa0-447f-84a6-9048b00c4bf7" />

Using an enum type will incorrectly allow TS users to do

```ts
const transitionTypeUppercase: string = chrome.history.TransitionType[someActualTransitionTypeValue]
```

when that will actually come out as undefined.